mgowrapper.go

```diff
@@ -0,0 +1,213 @@
+package mgohttp
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	opentracinglog "github.com/opentracing/opentracing-go/log"
+	mgo "gopkg.in/mgo.v2"
+	bson "gopkg.in/mgo.v2/bson"
+)
+
+type tracedMgoSession struct {
+	sess *mgo.Session
+	ctx  context.Context
+}
+
+func (ts tracedMgoSession) DB(name string) MongoDatabase {
+	sp := opentracing.SpanFromContext(ts.ctx)
+	sp.SetTag("db-name", name)
+	return tracedMgoDatabase{
+		db:  ts.sess.DB(name),
+		ctx: opentracing.ContextWithSpan(ts.ctx, sp),
+	}
+}
+
+func (ts tracedMgoSession) Ping() error {
+	sp, _ := opentracing.StartSpanFromContext(ts.ctx, "ping")
+	defer sp.Finish()
+
+	return logAndReturnErr(sp, ts.sess.Ping())
+}
+
+type tracedMgoDatabase struct {
+	db  *mgo.Database
+	ctx context.Context
+}
+
+func (t tracedMgoDatabase) C(collection string) MongoCollection {
+	return tracedMgoCollection{
+		collectionName: collection,
+		collection:     t.db.C(collection),
+		ctx:            t.ctx,
+	}
+}
+
+func (t tracedMgoDatabase) Run(cmd interface{}, result interface{}) error {
+	sp, _ := opentracing.StartSpanFromContext(t.ctx, "run")
+	defer sp.Finish()
+	sp.LogKV(opentracinglog.String("cmd", fmt.Sprintf("%#v", cmd)))
+
+	return logAndReturnErr(sp, t.db.Run(cmd, result))
+}
+
+type tracedMgoCollection struct {
+	collectionName string
+	collection     *mgo.Collection
+	ctx            context.Context
+}
+
+func (tc tracedMgoCollection) Update(selector interface{}, update interface{}) error {
+	sp, _ := opentracing.StartSpanFromContext(tc.ctx, "update")
+	sp.SetTag("collection", tc.collectionName)
+	sp.LogFields(queryToFields("selector", selector)...)
```

Comment:
Do we need to worry about sensitive data being logged?

Reply:
yes, but we no longer will - https://github.com/Clever/mgohttp/pull/7/commits/84e808d1d3a55506bbe3c3bed2b4a0215202866a